pkg/frontend/v2/frontend_scheduler_adapter.go

```diff
@@ -67,30 +66,31 @@ func (a *frontendToSchedulerAdapter) extractAdditionalQueueDimensions(
 		return nil, err
 	}
 
-	reqValues, err := util.ParseRequestFormWithoutConsumingBody(httpRequest)
-	if err != nil {
-		return nil, apierror.New(apierror.TypeBadData, err.Error())
-	}
-
 	switch {
-	case querymiddleware.IsRangeQuery(httpRequest.URL.Path):
-		start, end, _, err := querymiddleware.DecodeRangeQueryTimeParams(&reqValues)
+	case querymiddleware.IsRangeQuery(httpRequest.URL.Path), querymiddleware.IsInstantQuery(httpRequest.URL.Path):
+		decodedRequest, err := a.codec.DecodeMetricsQueryRequest(httpRequest.Context(), httpRequest)
+		if err != nil {
+			return nil, err
+		}
+		minT, err := decodedRequest.GetMinT()
 		if err != nil {
 			return nil, err
 		}
-		return a.queryComponentQueueDimensionFromTimeParams(tenantIDs, start, end, now), nil
-	case querymiddleware.IsInstantQuery(httpRequest.URL.Path):
-		time, err := querymiddleware.DecodeInstantQueryTimeParams(&reqValues, time.Now)
+		maxT, err := decodedRequest.GetMaxT()
 		if err != nil {
 			return nil, err
 		}
-		return a.queryComponentQueueDimensionFromTimeParams(tenantIDs, time, time, now), nil
+
+		return a.queryComponentQueueDimensionFromTimeParams(tenantIDs, minT, maxT, now), nil
```

Comment:
this is the actual bugfix - instead of using start/end or time for range and instant queries, we are determining this query component with actual minT and maxT for the query.